service/s3/endpoints.go

```diff
@@ -4325,3 +4239,602 @@ func (r *resolver) ResolveEndpoint(
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketIntelligentTieringConfigurationInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketInventoryConfigurationInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketLifecycleConfigurationInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketLoggingInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketMetricsConfigurationInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketNotificationConfigurationInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketOwnershipControlsInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketPolicyInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketReplicationInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketRequestPaymentInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketTaggingInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketVersioningInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutBucketWebsiteInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutObjectInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutObjectAclInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutObjectLegalHoldInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutObjectLockConfigurationInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutObjectRetentionInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutObjectTaggingInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *PutPublicAccessBlockInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *RestoreObjectInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *SelectObjectContentInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *UploadPartInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *UploadPartCopyInput) bindEndpointParams(p *EndpointParameters) {
+	p.Bucket = in.Bucket
+
+}
+
+func (in *WriteGetObjectResponseInput) bindEndpointParams(p *EndpointParameters) {
+
+	p.UseObjectLambdaEndpoint = ptr.Bool(true)
+}
+
+type resolveEndpointV2Middleware struct {
+	options  Options
```

Comment:
is this the client `Options` in `api_client.go`?
reason why im asking is because it seems that there was a previous effort to not pass the client `Options` around directly in middleware, but rather to create intermediary objects and only copy out what you need from the client `Options`
https://github.com/aws/aws-sdk-go-v2/blob/main/service/s3/api_client.go#L468
https://github.com/aws/aws-sdk-go-v2/blob/main/service/s3/api_op_GetObject.go#L517

would you be able to comment on whether you think that previous pattern should or should not be followed?

also, why give `resolver` its own field if it already exists in the same `options` field
```go
func addResolveEndpointV2Middleware(stack *middleware.Stack, options Options) error {
	return stack.Serialize.Insert(&resolveEndpointV2Middleware{
		options:  options,
		resolver: options.EndpointResolverV2,
	}, "ResolveEndpoint", middleware.After)
}
```

Reply:
Discussed offline - this is more important for exported middlewares or situations where code can't take a dependency on `Options` directly. For codegenned middleware internal to services, passing options at large is generally fine.